### PR TITLE
feat: allow specifying a resource map, closes #5844

### LINKED ISSUE #5844: [bug] cannot run sidecar/cmd from anywhere but src-tauri if it has shared libraries
### Describe the bug

running an executable that depends on other shared libraries won't work unless you place all the files/folders and the binary itself directly in the src-tauri folder and copy them as individual `resources`

### Reproduction

_No response_

### Expected behavior

should be able to run executables (with shared library deps) from a folder inside `src-tauri` otherwise the `src-tauri` directory becomes a huge mess

### Platform and versions

x86_64-pc-windows-msvc
tested in 1.2 and 2-alpha.0 both fail

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lucasfernog**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/bundler/src/bundle/settings.rs [hunk 1]
```diff
@@ -703,9 +709,14 @@ impl Settings {
...
-    match self.bundle_settings.resources {
-      Some(ref paths) => ResourcePaths::new(paths.as_slice(), true),
-      None => ResourcePaths::new(&[], true),
+    match (
+      &self.bundle_settings.resources,
+      &self.bundle_settings.resources_map,
+    ) {
+      (Some(paths), None) => ResourcePaths::new(paths.as_slice(), true),
+      (None, Some(map)) => ResourcePaths::from_map(map, true),
+      (Some(_), Some(_)) => panic!("cannot use both `resources` and `resources_map`"),
```
Comment (amrbashir) [inline tooling/bundler/src/bundle/settings.rs:747]: why panic here? seems like both can co-exist and merged into one final map

Comment (lucasfernog) [reply tooling/bundler/src/bundle/settings.rs:747]: Well the Tauri configuration does not allow both to be provided... so it's not worth implementing it until someone else is using the bundler.

Comment (amrbashir) [reply tooling/bundler/src/bundle/settings.rs:747]: cool

**Review by amrbashir — DISMISSED** [review]

**Review by amrbashir — DISMISSED** [review]
I updated the NSIS implementation and used `dunce` to simplify the path on WiX and NSIS

Other changes LGTM

**Review by amrbashir — DISMISSED** [review]

**Review by amrbashir — APPROVED** [review]

**Pull request merged**